mood_bot.py

```diff
@@ -0,0 +1,726 @@
+    elif power.upper() == 'ON':
+        target_time = mood_time_retriever(author)
+        auto_checker_power_switch = power.upper()
+    else:
+        target_time = 0
+        await auto_checker_error(context, power)
+    if isinstance(target_time, datetime.datetime):
+        while datetime.datetime.now() < target_time:
+            if auto_checker_power_switch == 'OFF':
+                break
+            await asyncio.sleep(1)
+        if auto_checker_power_switch == 'ON':
+            await mood_reset(context, False)
+            await mood_check(context)
+
+
+@mood_bot.command(name='status', help='Reports whether mood auto checker is on or off, '
+                                      'whether a mood check is in progress, '
+                                      'and what stage of the mood check process is open if applicable.')
+async def mood_status(context):
+    await context.send(f"The auto mood checker is {auto_checker_power_switch}.")
+    mood_check_status = 'has not been started.'
+    if date_gate == 'Closed':
+        mood_check_status = 'is in progress, '
+    await context.send(f"A mood check {mood_check_status}")
+    if date_gate == 'Closed':
+        gate_array = [date_gate, type_gate, rating_gate, why_gate]
+        await context.send(f"and is currently in the mood {stage_array[gate_array.index('Open')]} stage.")
+
+
+async def time_stamp_message(context, author):
+    time_stamp = mood_diary_retrieval(author, 'date', 'last_entry')
+    await context.send(f"You have an unfinished mood check from {time_stamp}.")
+
+
+def mood_stamp_maker(author):
+    mood_stamp = mood_diary_retrieval(author, 'type', 'last_entry')
+    return mood_stamp
+
+
+def rating_stamp_maker(author):
+    rating_stamp = mood_diary_retrieval(author, 'rating', 'last_entry')
+    return rating_stamp
+
+
+async def mood_rating_message(context, mood_stamp):
+    await context.send(f'You left off at rating. The mood type was {mood_stamp}')
+
+
+async def mood_why_message(context, mood_stamp, rating_stamp):
+    await context.send(f"You left off at the explanation. The mood type was {mood_stamp} "
+                       f"and the rating was {rating_stamp}.")
+
+
+@mood_bot.command(name='check', help='Run a mood check.')
+async def mood_check(context):
+    """This function starts a mood check by recording current time and date and asking the user how they feel."""
+    global date_gate, type_gate
+    author = context.author
+    if date_gate == 'Open':
+        entry_time = datetime.datetime.now()
+        mood_diary_storage(author, 'date', entry_time.strftime('%Y/%m/%d %H:%M:%S'))
+        await context.send("How are you feeling, " + str(author) + "?\n" +
+                           "Please select from the following mood types:\n" +
+                           str(mood_choices) + "\n" +
+                           "Please respond by typing 'mood_type' followed by your selected mood.")
+        date_gate = 'Closed'

```

Comment:
you can do gate +=1, and wrap it around somehow

Reply:
I have implemented the incrementing integer with the wraparound.